src/v/cluster/errc.h

```diff
@@ -79,6 +81,10 @@ struct errc_category final : public std::error_category {
             return "Invalid topic name";
         case errc::partition_not_exists:
             return "Requested partition does not exists";
+        case errc::partition_already_exists:
+            return "Requested partition already exists";
+        case errc::waiting_for_recovery:
+            return "Waiting for partition to recover";
```

Comment:
`errc::not_leader` is not handled here

Reply:
most likely not related to the PR, but anyway

Reply:
done